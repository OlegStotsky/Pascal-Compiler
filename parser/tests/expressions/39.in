var
	a, b : boolean;

begin
	a := true;
	b := false;
	a := a and b;
	a := a or b;
	a := a xor b;
end;