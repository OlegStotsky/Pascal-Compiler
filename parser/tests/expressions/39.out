BEGIN SYMTABLE
 VAR
  NAME : a
  TYPE : boolean
 VAR
  NAME : b
  TYPE : boolean
 TYPE : boolean
 TYPE : char
 TYPE : integer
 TYPE : float
END SYMTABLE
BEGIN STMT BLOCK
 BEGIN STMT ASSIGN
   a
  :=
   true
 END STMT ASSIGN
 BEGIN STMT ASSIGN
   b
  :=
   false
 END STMT ASSIGN
 BEGIN STMT ASSIGN
   a
  :=
    a
   and
    b
 END STMT ASSIGN
 BEGIN STMT ASSIGN
   a
  :=
    a
   or
    b
 END STMT ASSIGN
 BEGIN STMT ASSIGN
   a
  :=
    a
   xor
    b
 END STMT ASSIGN
END STMT BLOCK
